publish/kafka.go

```diff
@@ -0,0 +1,105 @@
+package publish
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/rotationalio/baleen/config"
+	"github.com/rotationalio/baleen/store"
+	"github.com/segmentio/kafka-go"
+)
+
+// A KafkaPublisher publishes objects to a Kafka topic.
+type KafkaPublisher struct {
+	conf     config.KafkaConfig
+	writer   *kafka.Writer
+	messages []kafka.Message
+}
+
+func New(conf config.KafkaConfig) (publisher *KafkaPublisher, err error) {
+	writer := &kafka.Writer{
+		Addr: kafka.TCP(conf.URL),
+	}
+	switch conf.Balancer {
+	case "RoundRobin":
+		writer.Balancer = &kafka.RoundRobin{}
+	case "LeastBytes":
+		writer.Balancer = &kafka.LeastBytes{}
+	case "Hash":
+		writer.Balancer = &kafka.Hash{}
+	case "Murmur2":
+		writer.Balancer = &kafka.Murmur2Balancer{}
+	case "CRC32":
+		writer.Balancer = &kafka.CRC32Balancer{}
+	default:
+		return nil, fmt.Errorf("unknown balancer specified: %s", conf.Balancer)
+	}
```

Comment:
Neat! Is this sort of boilerplate for Kafka or something custom?

Reply:
Basically trying to understand what options we have for balancers beyond the default.